test/lib/surt/surt_test.rb

```diff
@@ -0,0 +1,297 @@
+require 'test_helper'
+
+class SurtTest < ActiveSupport::TestCase
+  # Give us a nice syntax so we can visually align output vs. input
+  def assert_canonicalized(expected, url, message = nil, options = {})
+    assert_equal(expected, Surt.canonicalize(url, options), message)
+  end
+
+  test 'it does not modify canonical URLs' do
+    assert_canonicalized(
+      'http://archive.org/index.html',
+      'http://archive.org/index.html'
+    )
+  end
+
+  test 'it removes "www[digit?]" subdomains' do
+    assert_canonicalized(
+      'http://alexa.com/',
+      'http://www.alexa.com/',
+      'It did not remove a "www." subdomain'
```

Comment:
Is the "sign" of this message reversed? This message should state the _desired_ behavior, right? I could be confused by I don't think the "sign" of these messages are consistent within the test.

Reply:
The standard convention I’ve seen in the Ruby community is to use the message to say what went wrong (e.g. http://guides.rubyonrails.org/testing.html#your-first-failing-test) and use the test name to say what you expect to have happened. So I’ve generally been trying to follow that, but it looks like this is the only I did that way in this set of tests :\  (See, for example, https://github.com/edgi-govdata-archiving/web-monitoring-db/blob/master/test/models/annotation_test.rb)

Hmmm, which way do you want to go here?

Reply:
That is the convention I am used to as well. What confused me is that the two messages in this test choose opposite ways: 'It did not remove a "www." subdomain' and 'It removed a "www[digit]." subdomain'. Let's aim for the standard convention as you described it above and apply it in this PR. I'm not too concerned about going back and fixing places where this is reversed because the code, at least, is clear, and we have bigger things to do. Sound reasonable? I hope I'm bike-shedding here; if you think I am feel free to say so. :- )

Reply:
Huh, on further review, it looks like I was just all over the place here. Half the messages are sensible and half not. Not sure where my brain was when I did those tests :(

Updated them all to be (what failed). Also changed several use the language “it failed to…” since that seems more clear.